### ISSUE TITLE
[no-unused-vars] Inconsisitent behaviour with shadowed type name

### ISSUE THREAD

**Issue by fa93hws** [typescript-eslint/typescript-eslint#696]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": ["error"]
  }
}
```

```TS
import { SomeType } from 'foo';
export function fn0(SomeType: SomeType){}

type AnotherType;
export function fn1(AnotherType: AnotherType){}

export function fn2(NonExistedType: NonExistedType) {}
```

**Expected Result**
All pass the rule or none of them pass the rule

**Actual Result**
Import is not OK, but the rest is fine even though the first parameter is not used.
When type node is visited, the parameter is marked as used..😰 see: https://github.com/typescript-eslint/typescript-eslint/issues/363

**Additional Info**
I know it's not possible to implement this rule without type information which will slow down the linting a lot.. But it should be possible to make the behavior consistent?

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `X.Y.Z` |
| `@typescript-eslint/parser`        | `X.Y.Z` |
| `TypeScript`                       | `X.Y.Z` |
| `ESLint`                           | `X.Y.Z` |
| `node`                             | `X.Y.Z` |
| `npm`                              | `X.Y.Z` |

**Comment by fa93hws** [comment]
Oh, I just saw https://github.com/typescript-eslint/typescript-eslint/pull/688

**Comment by bradzacher** [comment]
I'll leave this open to keep it tracked until #688 is merged.
